vignettes/magpie_emulator_for_remind.Rmd

```diff
@@ -105,8 +105,84 @@ replace_flat_fits("SSP2-NDC-Budg600/linear/data_postfit_SSP2-NDC-Budg600-IND-pre
+If you just want to test the generated emulators in REMIND, it might be easier to manually change the relevant files. The emulator coefficients are in the file `f30_bioen_price.cs4r`,  but to do this manually you must reproduce some of the postprocessing `mrremind` would do. 
+
+If you replaced flat fits, your emulator will be in (for example in a `SSP2-NPI-Base` scenario) `SSP2-NPI-Base/replaced-flat/f30_bioen_price_SSP2-NPI-Base_replaced_flat.cs4r`. The first data lines will look something like this. There's no variable name header, but they represent `year, region, scenario, coefficient, value`. There are two coefficients, `a` and `b`, representing the linear fits. 
+
+```
+2000,CAZ,SSP2-NPI-Base,a,1
+2000,CHA,SSP2-NPI-Base,a,0.172909809987918
+2000,EUR,SSP2-NPI-Base,a,0.141727536350427
+2000,IND,SSP2-NPI-Base,a,0.166571929509765
+2000,JPN,SSP2-NPI-Base,a,0.0824748054814729
+```
+
+### Replace NAs with artificial fits
+The first thing you must do is fix the NAs, replacing them with an artificial fit that prohibits biomass production for that region.
+
+```
+# Manually correct NAs in the cs4r since we're not going through mrremind at this time
+newfilepath <- "SSP2-NPI-Base/replaced-flat/f30_bioen_price_SSP2-NPI-Base_replaced_flat.cs4r"
+new <- read.magpie(newfilepath)
+
+new[,,"a"][is.na(new[,,"a"])] <- 1
+new[,,"b"][is.na(new[,,"b"])] <- 0.1
+
+write.magpie(new, newfilepath)
+```
+
+### Use the file in REMIND as base
+
+The linear coefficients file in REMIND is in the `path/to/remind/modules/30_biomass/magpie_40/input/f30_bioen_price.cs4r` in the REMIND folder. It has an additional `rcp` dimension, and contains data for several other scenarios, e.g.:
+
+```
+* description: coefficients for the bioenergy supplycurve
+* unit: none
+* comment: Data aggregated (toolAggregate): Thu Dec  2 19:46:36 2021
+* origin: calcOutput(type = "BiomassPrices", file = "f30_bioen_price.cs4r", round = 6) (madrat 2.7.1 | mrremind 0.92.2)
+* creation date: Mon Dec  6 09:33:08 2021
+2000,LAM,SDP,rcp45,a,0.124899
+2000,OAS,SDP,rcp45,a,0.175994
+2000,SSA,SDP,rcp45,a,0.207542
+2000,EUR,SDP,rcp45,a,0.074097
+2000,NEU,SDP,rcp45,a,0.125951
+2000,MEA,SDP,rcp45,a,0.329863
+2000,REF,SDP,rcp45,a,0.100725
+```
+
+The header doesn't make a difference in this case. If you want to force REMIND to use your new emulators, all you have to do is replace all scenarios and rcps in this file with fits from your new emulator. First make copy of the original to be sure.
+
+`cp path/to/remind/modules/30_biomass/magpie_40/input/f30_bioen_price.cs4r` `path/to/remind/modules/30_biomass/magpie_40/input/f30_bioen_price_old.cs4r`
+
+Then in the same R script before do:
+
+```
+# Now we add this data to an old file with the correct structure 
+oldfilepath <- "/path/to/remind/modules/30_biomass/magpie_40/input/f30_bioen_price_old.cs4r"
+old <- read.magpie(oldfilepath)
+
+# Create a copy magclass object
+out <- old
+
+# Replaces repeating data on all dimensions
+out[,,] <- new
```

Comment:
As you write below this replaces *all* scenarios with the single new one. But is this what we want? In your current case, for example, you calculated the supply curves just for SSP2-Base. Shouldn't we then only replace the "SSP2, none" case in the "old" file and leave all other supply curves untouched? Otherwise REMIND would use SSP2-Base emulators (which tend to have low bioenergy prices) also for policy runs, where bioenergy usually is much more expensive.

Reply:
Good point, I updated it with a couple of more specific examples. But doesn't the choice of emulator depend only on SSP and RCP? So policy and Base runs end up using the same emulators?

Reply:
Yes, the choice of the emulator depends on SSP and RCP. And the different climate targets (Base, NDC, PeakBudgX, PeakBudgY, ...) also come with different RCP switches (`cm_rcp_scen`) in REMIND. Thus, the different policy runs use different emulators. Therefore, in the end we will need to calculate a bunch of new emulators for the most common SSP-RCP combinations.